src/main/scala/nl.knaw.dans.easy.deposit/StagedFilesTarget.scala

```diff
@@ -31,30 +31,52 @@ import nl.knaw.dans.bag.ImportOption.{ ATOMIC_MOVE, COPY, ImportOption, MOVE }
 case class StagedFilesTarget(draftBag: DansBag, destination: Path) extends DebugEnhancedLogging {
 
   /**
+   * Moves files from stagingDir to draftBag if no files in the bag would be overwritten.
+   *
    * @param stagingDir the temporary container for files, unique per request, same mount as draftBag
    * @return
    */
   def takeAllFrom(stagingDir: File): Try[Any] = {
-    logger.info(s"moving from staging [$stagingDir] to ${ draftBag.baseDir / destination.toString }")
+    // read files.xml at most once, not at all if the first file appears to exist as payload
+    lazy val fetchFiles = draftBag.fetchFiles.map(_.file)
 
-    def moveToDraft(file: File): Try[Unit] = {
-      val bagRelativePath = destination.resolve(stagingDir.relativize(file))
-      // handle fetch items files just in case pruning gets implemented, for example after rejecting or un-publishing
+    def moveNewFiles = {
+      logger.info(s"moving from staging [$stagingDir] to ${ draftBag.baseDir / destination.toString }")
+      val (triesOfSourceTarget, duplicates) = stagingDir
+        .list
+        .map(sourceToTarget)
+        .partition(x => x.isSuccess)
+      if (duplicates.nonEmpty) collectExistingFiles(duplicates)
+      else {
+        // with triesOfSourceTarget.toStream.map(...)..find(_.isFailure).getOrElse(draftBag.save)
+        // all is still well
+        // with triesOfSourceTarget.toStream.map(...).failFastOr(draftBag.save)
+        // only one file is saved in files.xml
+        // TODO better test fore DataFiles.delete
```

Comment:
@rvanheest 
This difference between getOrElse here and the pattern match in failFastOr is puzzling me.

Reply:
Perhaps the argument should somehow be made lazy?